src/main/index.ts

```diff
@@ -97,11 +98,43 @@ async function start(bootBg: boolean) {
   }
 }
 
+
+if (process.defaultApp) {
+  if (process.argv.length >= 2) {
+    app.setAsDefaultProtocolClient('urbit-port', process.execPath, [path.resolve(process.argv[1])])
+  }
+} else {
+  app.setAsDefaultProtocolClient('urbit-port')
+}
+
+const gotTheLock = app.requestSingleInstanceLock()
+
+if (!gotTheLock) {
+  app.quit()
+  return;
+}
+
+app.on('second-instance', (event, commandLine, workingDirectory) => {
+  // Someone tried to run a second instance, we should focus our window.
```

Comment:
I believe you need to add the same protocol handling here for Windows or Linux, because only Mac fires the 'open-url' event. You can see an example in `main-window.ts` and in fact we probably should merge these at some point.